src/ledstrip.py

```diff
@@ -813,6 +823,7 @@ def _sunrise_sunset(self, seconds=None, milliseconds=None, setting=True):
         for temp in xrange(temp_0,temp_n,temp_step):
             if self._sequence_stop_signal: #Bail if sequence should stop
                 return None
+            print(temp)
             k = u"%sk" % temp
             self.fade(k, fade_time=((100+z_factor)/(65-x_step)), check=check) #ms, slows down as sunset progresses
```

Comment:
This will put the timing function out.

The sunset is an exponential fade, slowing as the sun goes down. So if you're starting from a lower temperature, there will be fewer steps, thus it will happen faster than the intended 1min / 30 min / 1 hr / 2hr etc.

What we need to do is modify the z_factor to account for the smaller colour steps. 

That said, your code is already an improvement because now we can go from any colour we like!